great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -0,0 +1,496 @@
+            except ValueError:
+                success = False
+
+        return {
+            'success': success
+        }
+
+    ###
+    ###
+    ###
+    #
+    # Column Map Expectation Implementations
+    #
+    ###
+    ###
+    ###
+
+    @DocInherit
+    @MetaSqlAlchemyDataSet.column_map_expectation
+    def expect_column_values_to_be_null(self,
+        column,
+        mostly=None,
+        result_format=None, include_config=False, catch_exceptions=None, meta=None
+    ):
+
+        return sa_column(column) != None
+
+    @DocInherit
+    @MetaSqlAlchemyDataSet.column_map_expectation
+    def expect_column_values_to_not_be_null(self,
+        column,
+        mostly=None,
+        result_format=None, include_config=False, catch_exceptions=None, meta=None
+    ):
+
+        return sa_column(column) == None
+
+
+    @DocInherit
+    @MetaSqlAlchemyDataSet.column_map_expectation
+    def expect_column_values_to_be_in_set(self,
+        column,
+        values_set,
+        mostly=None,
+        result_format=None, include_config=False, catch_exceptions=None, meta=None
+    ):
+        return not_(sa_column(column).in_(tuple(values_set)))
+
+    @DocInherit
+    @MetaSqlAlchemyDataSet.column_map_expectation
+    def expect_column_values_to_be_between(self,
+        column,
+        min_value=None,
+        max_value=None,
+        allow_cross_type_comparisons=None,
+        parse_strings_as_datetimes=None,
+        mostly=None,
+        result_format=None, include_config=False, catch_exceptions=None, meta=None
+    ):
+        if parse_strings_as_datetimes is not None:
+            raise ValueError("parse_strings_as_datetimes is not currently supported in SqlAlchemy.")
+
+        if min_value > max_value:
+            raise ValueError("min_value cannot be greater than max_value")
+
+        if min_value is None and max_value is None:
+            raise ValueError("min_value and max_value cannot both be None")
+
+        return not_(and_(
+                    min_value <= sa_column(column),
+                    sa_column(column) <= max_value
+            ))
+
+
+    ###
+    ###
+    ###
+    #
+    # Column Aggregate Expectation Implementations
+    #
+    ###
+    ###
+    ###
+    @DocInherit
+    @MetaSqlAlchemyDataSet.column_aggregate_expectation
+    def expect_column_max_to_be_between(self,
+        column,
+        min_value=None,
+        max_value=None,
+        parse_strings_as_datetimes=None,
+        output_strftime_format=None,
+        result_format=None, include_config=False, catch_exceptions=None, meta=None
+    ):
+
+        if min_value is None and max_value is None:
+            raise ValueError("min_value and max_value cannot both be None")
+
+        if parse_strings_as_datetimes:
+            raise ValueError("parse_strings_as_datetimes is not supported in SqlAlchemy")
+
+        col_max = self.engine.execute(
+            select([sa_func.max(sa_column(column))]).select_from(table(self.table_name))
```

Comment:
Got it. So `column_aggregate_expectations` act almost exactly like they do in `PandasDataSet`.

Side thought: we need to update our docs for custom expectations.